strapi/strapi
# Improved editing of dynmaic zones by adding toggle for visiblility and

![strapi_dz_improvement](https://user-images.githubusercontent.com/1172672/113051968-e0f9d480-91a6-11eb-97a8-481c071d396f.gif)


### What does it do?

- Moved dynamic zone CTAs into single bar so every new added CTA will be positioned without the need of absolute positioning
- Added expand function to dynamic zone components by just showing the upper part of compontent for easier idenfication
- Added expand all function to dynamic zones
- Changed move up and move down to single function with offset to reduce nearly duplicated code
- Added move to top and move to bottom button for dynamic zones

### Why is it needed?

1. Editing a content type with many elements in a dynamic zone is currenlty very confusing.
By just showing the first parts of a compontent you could see the top sorted elements first which ist mostly a label or something else which should be enough to identify the compontent you want to edit.

2. Improving moving elements in dynamic zones because new elements could only be added at the end of the zone. So an action with "move to top" will speed up editing dynamic zones.

3. Every single action button for a dynamic zone component is set absolute. To add a new cta you will always need to repositioning every single element.


### How to test it?

Create a content type with a dynamic zone.

_Expand elements:_
Add elements to the dynamic zone and each element will show an eye to toggle the visibility. The dynamic zone itself will have an eye too for toggling all elements at once.  
Default is always showing all components.

_Move elements:_
Add more than 2 items and every item, should have an "move to top" and/or "move to bottom" button.

### Related issue(s)/PR(s)

Let us know if this is related to any issue/pull request

Currently i can't find any issue about this function so its just an improvement of editing dynamic zones.
The nearest issue i found was #7941 so i have added the "move to top" and "move to bottom" too.
